assert/assertions_test.go

```diff
@@ -903,3 +903,18 @@ func TestNotZero(t *testing.T) {
 		True(t, NotZero(mockT, test, "%#v is not the %v zero value", test, reflect.TypeOf(test)))
 	}
 }
+
+func TestJSONEq(t *testing.T) {
```

Comment:
Please, split the test in different test cases so it's clear all the different tests that are being tested and when something fails we can know what's the issue from the failing test.